src/ripple/app/misc/Validations.cpp

```diff
@@ -458,20 +467,25 @@ class ValidationsImp : public Validations
 
         mWriting = true;
         app_.getJobQueue ().addJob (
-            jtWRITE, "Validations::doWrite",
-            [this] (Job&) { doWrite(); });
+            jtWRITE, "Validations::queuedWrite",
+            [this] (Job&) { queuedWrite(); });
     }
 
-    void doWrite ()
+    void queuedWrite ()
     {
         auto event = app_.getJobQueue ().getLoadEventAP (jtDISK, "ValidationWrite");
 
-        std::string insVal ("INSERT INTO Validations "
+        ScopedLockType sl (mLock);
+        doWrite (sl);
+    }
+
+    void doWrite (ScopedLockType& sl)
```

Comment:
Passing in the reference to the lock is not required by the functionality.  It's a way of signaling to maintainers that the lock must be held when `doWrite()` is called.  I suppose a comment to that effect would be useful as well 😃.